pymongo/uri_parser.py

```diff
@@ -506,22 +510,23 @@ def parse_uri(
         host_part = path_part
         path_part = ""
 
-    if not path_part and "?" in host_part:
-        raise InvalidURI("A '/' is required between the host list and any options.")
-
     if path_part:
         dbase, _, opts = path_part.partition("?")
-        if dbase:
-            dbase = unquote_plus(dbase)
-            if "." in dbase:
-                dbase, collection = dbase.split(".", 1)
-            if _BAD_DB_CHARS.search(dbase):
-                raise InvalidURI('Bad database name "%s"' % dbase)
-        else:
-            dbase = None
+    else:
+        # There was no slash in scheme_free, check for a sole "?".
+        host_part, _, opts = host_part.partition("?")
+
+    if dbase:
```

Comment:
`dbase` will always be `None` for slashless URIs. Is that expected?

Reply:
Yes you can't specify a db without a slash. EG:
```
mongodb://example.com/admin?tls=True
```
without the slash is invalid:
```
mongodb://example.comadmin?tls=True
```

Reply:
Got it, looks good then!